subresource-loading.md

```diff
@@ -237,20 +244,38 @@ When returning to the event loop, if there are any chunk IDs on the list to fetc
 - When the fetch returns from the server, map in all fetched chunk IDs and add them to the persistent cache.
 - Then, the original fetch can continue at the top of the algorithm. It should either be served by a mapped-in chunk or 404.
 
+#### Pros
+* Mixing in decelration of the resources in the bundle and scopes covered by it in ways that enable use of both techniques, without requiring hard-to-implement [Cache Digests](https://tools.ietf.org/html/draft-ietf-httpbis-cache-digest-05).
+* Decouples the resource number from the number of chunks in ways that enable higher scalability in cases of closely knit, very small modules.
+* Update flow doesn't require content-addressable URLs, which avoids problems with unnecessary cache invalidation of references, as well as "purge" like semantics when resource version changes.
+  - Unclear if that part would work properly in the "online URL enforcement" revalidation flow.
+
+#### Cons
+* Requires declaring full dependency map on each entry point, that includes all the chunks required by all the entry points. If developers will get that wrong, it can result in spurious downloads.
+* A "just download the entire bundle" fallback is not feasible, since the full bundle can be extremely large. Related, caches that ignores Vary will have have A Bad Time™. 
+* As the downloaded resources are likely to be significantly smaller than the the complete bundle, [compression benefits of subset recompression](https://dev.to/riknelix/fast-and-efficient-recompression-using-previous-compression-artifacts-47g5) are likely to be small. (See [results](https://dev.to/riknelix/fast-and-efficient-recompression-using-previous-compression-artifacts-47g5#results) for 90% removal)
+* It's improbable that all entry points require the same order of resources. This proposal doesn't provide a way for bundlers to address this and enable different chunk order for different entry points.
```

Comment:
I'm not sure what you mean; this proposal lets bundlers order their resource bundle output (both which order the chunks come in, and the order of the individual resources within the chunks). Servers could preserve the order that the bundler set.

Reply:
Let's say that:
`a.js` depends on resources C, D, G, H, K, L (in that order)
`b.js` depends on resources G, F, L, P, D, C (in that order)

What order would a bundler output when creating a mega bundle that's supposed to serve both those entry points?

Reply:
Ah, I was imagining that the request from the client to the server would be treated as an unordered set, and the server would tend to respond with the chunks in its fixed order. (So this would be more streaming-friendly as well.)

Reply:
In that case, we're not subsetting bundles but reordering them, and any compression benefits of reusing previous high-quality compression would be forfeit.

Reply:
I'm confused; I thought I was describing subsetting. The server would have the whole bundle; the client would incate the parts it wants (interpreted unordered); and the server woudl respond with whatever order the server decides. Where's the reordering?

Reply:
Essentially, the order in which the server would ideally response is likely to vary depending on the entry point to which it is responding. For optimal compression gains, we want the server to have the order ready (and compressed) in advance, and then subset that dynamically, but without reordering it. Does that help to clarify?

Reply:
Right, I'm saying, I agree that the server should not have to reorder things.